m3u8/parser.py

```diff
@@ -447,7 +447,7 @@ def _parse_byterange(line, state, **kwargs):
 
 
 def _parse_targetduration(**parse_kwargs):
-    return _parse_simple_parameter(cast_to=float, **parse_kwargs)
```

Comment:
I don't think I understand how this rounds up rather than down?

Reply:
I understood the problem does not happen when `dumps` get called. It actually happens when the playlist is parsed. `target_duration` was indeed being converted to float (as the diff shows)

Reply:
Oh, I see - in the `tests/playlists.py` change, you're updating the test data directly.

Reply:
Yes! Do you think that’s a bad approach? I changed the test playlist because, after reading the specifications, I believe a playlist should not have a target duration as a float.

Reply:
I think it's OK. At first I misunderstood the changes in this PR - I thought the test change was reflecting a new output behavior, but it was changed to reflect what's legal in the spec.